### DISCUSSION TITLE
Get reference to newly created window in script

### DISCUSSION THREAD

**Discussion by raiguard** [qtile/qtile#3521] [Q&A] [answered]
I am attempting to dive into scripting qtile. The documentation has been very confusing, but I finally figured out that all of the commands are listed in the qtile shell, and I figured out how to access the shell by sheer trial and error. Once I got in there and poked around, I figured out how to float a window and set a size and position for it.

However, what I need to do is create a _new_ window from a script, and then set the properties on that window. The only function I've found so far is `root.spawn()` to run a command. However, it does not appear to return any sort of reference to the window.

How can I do this?

(X/Y problem: I wish to create a floating popup window in the middle of the screen)

```python
from libqtile.command.client import InteractiveCommandClient
c = InteractiveCommandClient()

window = c.spawn("kitty lazygit")
window.toggle_floating()
window.set_size_floating(600, 400)
window.set_position_floating(3540, 520)
```

**Comment by SweepingBishops** [comment, marked as answer]
Consider adding it to the float rules in the qtile config. It accepts a list of `Matches`. 
Open kitty with the `class` parameter to set a custom class name.
```
floating_layout = layout.Floating(float_rules=[
    # Run the utility of `xprop` to see the wm class and name of an X client.
    *layout.Floating.default_float_rules,
    Match(wm_class='ssh-askpass'),  # ssh-askpass
    Match(title='pinentry'),  # GPG key password entry
    Match(wm_class='galculator'),   # Calculator
    ])
```
Set the window width and height in the kitty config. I don't know how to set the window position, you _could_ drag it . But there surely is a way to tell qtile where to open the window, I just don't know it.

**Reply by raiguard** [reply]
Thank you. I actually ended up switching to Sway a day after making this post, and I ended up using a similar solution there as well.
